internal/pkg/instrumentation/bpf/net/http/client/bpf/probe.bpf.c

```diff
@@ -6,9 +6,16 @@
 
 char __license[] SEC("license") = "Dual MIT/GPL";
 
-#define MAX_HOSTNAME_SIZE 256
+#define MAX_HOSTNAME_SIZE 128
 #define MAX_PROTO_SIZE 8
 #define MAX_PATH_SIZE 100
+#define MAX_SCHEME_SIZE 8
+#define MAX_OPAQUE_SIZE 8
+#define MAX_RAWPATH_SIZE 8
+#define MAX_RAWQUERY_SIZE 128
+#define MAX_FRAGMENT_SIZE 50
```

Comment:
I think it is preferred to have these sizes as powers of multiples of 8 if possible, to avoid padding issues which can cause the pref buffer reader to fail to read the entries successfully.

Reply:
sounds good, I'll update. one thing with this approach though is that we're hitting the 512 byte stack limit pretty easily and can't support very long values for a lot of these fields. would like to open a follow up after this to refactor this section